### PR TITLE
Fix #22: Handle JSON Celery task serialization

### LINKED ISSUE #22: Celery backend: Handle other serializers than pickle (JSON)
When switching to the Celery back-end I have an issue: The parameters for the task, "emails" are instances of django.core.mail.message.EmailMessage or EmailMultiAlternatives. There are both not serializable to JSON. We do not use pickle as recommended by Celery's documentation.

I will look at it tomorrow but in the meantime if you have any idea to fix it, thanks for sharing.

### PR DESCRIPTION
**Pull request by davidfischer-ch**
Work in progress.

### CODE REVIEW

File: djmail/backends/celery.py [hunk 1]
```diff
@@ -15,4 +18,6 @@ def _send_messages(self, email_messages):
         if len(email_messages) == 0:
             return 0
 
+        if settings.CELERY_TASK_SERIALIZER in ('json', ):
```
Comment (davidfischer-ch) [inline djmail/backends/celery.py]: Testing with in because I expect we can add YAML ...

File: djmail/tests.py [hunk 2]
```diff
@@ -250,9 +238,18 @@ def test_simple_send_email_with_magic_builder_1_with_low_priority(self):
...
-class SerializationEmailTests(TestCase):
-    def setUp(self):
-        models.Message.objects.all().delete()
+class SerializationEmailTests(EmailTestCaseMixin, TestCase):
+    def test_serialization_loop(self):
+        data = utils.serialize_email_message(self.email)
+        email_bis = utils.deserialize_email_message(data)
+        self.assertEmailEqual(self.email, email_bis)
+
+    def test_json_serialization_loop(self):
```
Comment (davidfischer-ch) [inline djmail/tests.py]: I'd like to test with Celery but strangely the test that should fail (JSON serialization of task), even if I override CELERY_TASK_SERIALIZER, passed! This is not the case in production ...

**Comment by davidfischer-ch** [comment]
That issue with Travis again. Pull request is ready for review.

File: djmail/utils.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+from django.utils.encoding import force_bytes
+try:
+    # Django >= 1.4.5
+    from django.utils.encoding import force_bytes, force_text
+except ImportError:
+    # Django < 1.4.5
+    from django.utils.encoding import force_bytes, smart_unicode as force_text
+from django.utils.six import string_types
```
Comment (bameda) [inline djmail/utils.py]: You are importing ```from django.utils.encoding import force_bytes``` twice (lines ```13``` and ```16```) and this (```force_bytes```) causes the error in Travis

**Comment by davidfischer-ch** [comment]
The force_bytes is new in Django 1.5 and I do not want to copy-paste it in this project.
Can't we simply drop Django 1.4 compatibility?

FYI Releases up-to 1.8 are officially deprecated for security reasons.

**Comment by bameda** [comment]
Can you use ```smart_byte``` instead? 
```
from django.utils.encoding import smart_bytes as force_bytes
```
If that doesn't work we can deprecate django 1.4

**Comment by bameda** [comment]
Deprecated, see https://github.com/bameda/djmail/pull/30

**Pull request closed without merging**